aws/resource_aws_dx_public_virtual_interface_test.go

```diff
@@ -18,7 +20,66 @@ func TestAccAwsDxPublicVirtualInterface_basic(t *testing.T) {
 	if connectionId == "" {
 		t.Skipf("Environment variable %s is not set", key)
 	}
-	vifName := fmt.Sprintf("terraform-testacc-dxvif-%s", acctest.RandString(5))
+
+	var vif directconnect.VirtualInterface
+	resourceName := "aws_dx_public_virtual_interface.test"
+	rName := fmt.Sprintf("tf-testacc-public-vif-%s", acctest.RandString(10))
+	// DirectConnectClientException: Amazon Address is not allowed to contain a private IP
```

Comment:
Thank you for these comments

Reply:
I added a function to generate a random IP but hit these various conditions so ended up hard-coding. I ensured that there is no conflict for between any of the `aws_dx_public_virtual_interface` and `aws_dx_hosted_public_virtual_interface` tests.
I submitted a PR - https://github.com/hashicorp/terraform/pull/22312 - for that function as it is useful in other scenarios.